### DISCUSSION TITLE
Feature Request: Add a contains_not_null method

### DISCUSSION THREAD

**Discussion by xiaozhuai** [nlohmann/json#4245] [Ideas]
```cpp
if (obj.contains("key") && !obj.at("key").is_null()) {
    // do something
}
```
We check `obj.contains("key") && !obj.at("key").is_null()` very often, could we add a method,
for example named `contains_not_null`, then we can simplify the code.
```cpp
if (obj.contains_not_null("key")) {
    // do something
}
```

**Comment by nlohmann** [comment]
This is actually a good idea! There is also the idea of #2715. Let me think about it.
